### PR TITLE
Use list(kwargs) instead of list(kwargs.keys)

### LINKED ISSUE #7198: Typo in raise from einsum
### Description

In this line:

https://github.com/cupy/cupy/blob/master/cupy/linalg/_einsum.py#L511

einsum() tries to raise an error:

```py
raise TypeError('Did not understand the following kwargs: %s'
                        % list(kwargs.keys))
```

but I believe it should be `list(kwargs.keys())` or simply `list(kwargs)`.

### To Reproduce

```py
>>> cp.einsum('i', cp.ones(7), asdf=42)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/camp/jensj/venv/lib64/python3.9/site-packages/cupy/linalg/_einsum.py", line 512, in einsum
    % list(kwargs.keys))
TypeError: 'builtin_function_or_method' object is not iterable
```

### Installation

_No response_

### Environment

```
OS                           : Linux-4.18.0-372.26.1.el8_6.x86_64-x86_64-with-glibc2.28
Python Version               : 3.9.13
CuPy Version                 : 11.3.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.23.5
SciPy Version                : 1.9.3
Cython Build Version         : 0.29.32
Cython Runtime Version       : None
CUDA Root                    : /usr/local/cuda
nvcc PATH                    : /usr/local/cuda/bin/nvcc
CUDA Build Version           : 11080
CUDA Driver Version          : 11080
CUDA Runtime Version         : 11080
cuBLAS Version               : (available)
cuFFT Version                : 10900
cuRAND Version               : 10300
cuSOLVER Version             : (11, 4, 1)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 8)
Thrust Version               : 101501
CUB Build Version            : 101501
Jitify Build Version         : <unknown>
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : Quadro RTX 4000
Device 0 Compute Capability  : 75
Device 0 PCI Bus ID          : 0000:21:00.0
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by jjmortensen**
Closes #7198

### CODE REVIEW

**Review by kmaehashi — DISMISSED** [review]
~Thanks, LGTM!~

File: cupy/linalg/_einsum.py [hunk 1]
```diff
@@ -509,7 +509,7 @@ def einsum(*operands, **kwargs):
         optimize = 'greedy'
     if kwargs:
         raise TypeError('Did not understand the following kwargs: %s'
-                        % list(kwargs.keys))
+                        % list(kwargs))
```
Comment (kmaehashi) [inline cupy/linalg/_einsum.py:512]:
```suggestion
                        % list(kwargs.keys()))
```

Comment (jjmortensen) [reply cupy/linalg/_einsum.py]: OK, changed it. Technically, `list(kwargs)` and `list(kwargs.keys())` will give the same result - a list of keys.

**Review by kmaehashi — APPROVED** [review]

File: cupy/linalg/_einsum.py (hunk 1 above)
Comment (kmaehashi) [reply cupy/linalg/_einsum.py]: Ah you're right 😄

**Pull request merged**